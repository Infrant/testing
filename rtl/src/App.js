// import {useEffect, useState} from "react";
//
// import './App.css';
// import {Users} from "./users/users";
//
// function App() {
//     const [data, setData] = useState(null)
//     const [toggle, setToggle] = useState(false)
//     const [inputValue, setInputValue] = useState('')
//
//     const handleToggle = () => setToggle(prev => !prev)
//
//     useEffect(() => {
//         setTimeout(() => {
//             setData({})
//         }, 100)
//     }, [])
//
//     return (
//         <div className="App">
//             {data && <div style={{color: 'green'}}>data</div>}
//             <h1>Title</h1>
//             <button onClick={handleToggle} data-testid={'toggle-btn'}>click</button>
//             <input
//                 type="text"
//                 placeholder={'text input1'}
//                 data-testid={'input-elem'}
//                 value={inputValue}
//                 onChange={e => setInputValue(e.target.value)}
//             />
//             {toggle && <div data-testid={'toggle-div'}>Toggle div</div>}
//             <div data-testid={'input-value-elem'}>{inputValue}</div>
//             <Users/>
//         </div>
//     );
// }
//
// export default App;

import React from 'react';
import {BrowserRouter, Routes, Route, Link} from "react-router-dom";
import MainPage from "./pages/mainPage";
import AboutPage from "./pages/aboutPage";
import ErrorPage from "./pages/errorPage";
import {Users} from "./users/users";
import UserDetailsPage from "./pages/userDetailsPage";
import AppRouter from "./router/AppRouter";
import NavBar from "./components/navBar/navBar";

const App = () => {
    return (
        <div>
            <NavBar/>
            <AppRouter/>
        </div>

    );
};

export default App;
