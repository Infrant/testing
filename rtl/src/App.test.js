import {render, screen, fireEvent, act} from '@testing-library/react';
import userEvent from "@testing-library/user-event";

import App from './App';
import {MemoryRouter} from "react-router-dom";

describe('App', () => {
    beforeEach(() => {
        render(
            <MemoryRouter>
                <App/>
            </MemoryRouter>
        );
    })
    test('renders learn react link', () => {
        const titleElem = screen.getByText(/title/i);
        const btnElem = screen.getByRole('button');
        const inputElem = screen.getByPlaceholderText('text input1');
        expect(titleElem).toBeInTheDocument();
        expect(btnElem).toBeInTheDocument();
        expect(inputElem).toBeInTheDocument();
        expect(inputElem).toMatchSnapshot();
    });

    test('async', async () => {
        // const titleElem = screen.queryByText(/title2/i)
        // expect(titleElem).toBeNull()

        screen.debug()
        const dataElem = await screen.findByText(/data/i)
        expect(dataElem).toBeInTheDocument()
        expect(dataElem).toHaveStyle({color: 'green'})
        screen.debug()
    })

    test('click event', () => {
        const btnElem = screen.getByTestId('toggle-btn')
        expect(screen.queryByTestId('toggle-div')).toBeNull()
        fireEvent.click(btnElem)
        expect(screen.queryByTestId('toggle-div')).toBeInTheDocument()
        fireEvent.click(btnElem)
        expect(screen.queryByTestId('toggle-div')).toBeNull()
    })

    test('input event', () => {
        const inputElem = screen.getByTestId('input-elem')
        expect(screen.queryByTestId('input-value-elem')).toHaveTextContent('')
        // fireEvent.change(inputElem, {
        //     target: {
        //         value: 'value for test'
        //     }
        // })
        act(() => {
            userEvent.type(inputElem, 'value for test')
        })
        expect(screen.queryByTestId('input-value-elem')).toHaveTextContent('value for test')
    })
})
