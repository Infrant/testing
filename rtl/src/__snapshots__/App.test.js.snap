// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders learn react link 1`] = `
<input
  placeholder="text input1"
  type="text"
/>
`;
